src/modules/NtpClient.rb

```diff
@@ -266,92 +253,59 @@ def GetCountryNames
-        else
-          l_country = Ops.get(country_names, l_country, l_country)
-        end
-        if label != "" && l_country != ""
-          label = Builtins.sformat("%1 (%2, %3)", s, label, l_country)
-        elsif label == "" && l_country == ""
-          label = s
-        else
-          label = Builtins.sformat("%1 (%2%3)", s, label, l_country)
-        end
-
+      default = false
+      items = servers.map do |server, attrs|
         # Select the first occurrence of pool.ntp.org as the default option (bnc#940881)
-        if default_already_chosen
-          selected = false
-        else
-          selected = default_already_chosen = s.end_with?("pool.ntp.org")
-        end
+        selected = default ? false : default = server.end_with?("pool.ntp.org")
 
-        if terse_output
-          next Item(Id(s), s, selected)
-        else
-          next Item(Id(s), label, selected)
-        end
+        next Item(Id(server), server, selected) if terse_output
+
+        country_label = country.empty? ? country_names[attrs["country"]] || attrs["country"] : ""
+
+        label = server + country_server_label(attrs["location"].to_s, country_label.to_s)
+
+        Item(Id(server), label, selected)
       end
 
-      deep_copy(items)
+      items
     end
 
     # Read and parse /etc.ntp.conf
     # @return true on success
     def ProcessNtpConf
       if @config_has_been_read
-        Builtins.y2milestone("Configuration has been read already, skipping.")
+        log.info "Configuration has been read already, skipping."
         return false
       end
 
       conf = nil
-      if FileUtils.Exists("/etc/ntp.conf")
-        conf = Convert.to_map(SCR.Read(path(".etc.ntp_conf.all")))
-      end
+      conf = SCR.Read(path(".etc.ntp_conf.all")) if FileUtils.Exists("/etc/ntp.conf")
 
-      if conf == nil
-        Builtins.y2error(
-          "Failed to read /etc/ntp.conf, either it doesn't exist or contains no data"
-        )
+      if conf.nil?
+        log.error("Failed to read /etc/ntp.conf, either it doesn't exist or contains no data")
         return false
       end
-      Builtins.y2milestone("Raw ntp conf %1", conf)
+
+      log.info("Raw ntp conf #{conf}")
       @config_has_been_read = true
-      value = Ops.get_list(conf, "value", [])
+      value = conf["value"] || []
       index = -1
       @ntp_records = Builtins.maplist(value) do |m|
-        index = Ops.add(index, 1)
-        type = Ops.get_string(m, "name", "")
+        index += 1
+        type = m["name"].to_s
         address = Ops.get_string(m, "value", "")
         options = ""
-        if Builtins.contains(
-            [
-              "server",
-              "peer",
-              "broadcast",
-              "broadcastclient",
-              "manycast",
-              "manycastclient",
-              "fudge",
-              "restrict"
-            ],
-            type
-          )
+        if ["server", "peer", "broadcast", "broadcastclient", "manycast",
```

Comment:
Maybe you could define a constant before the method (are at least add a comment about what are these terms).